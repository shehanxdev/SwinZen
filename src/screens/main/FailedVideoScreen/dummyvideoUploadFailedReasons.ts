export const dummyvideoUploadFailedReasons = [
  { id: 1, message: 'Too far away from golfer' },
  { id: 2, message: 'Too close to golfer' },
  { id: 3, message: 'Poor lighting. (Usually when indoors or at dusk outside.)' },
  { id: 4, message: 'Too much clutter in background (Usually indoors)' },
  { id: 5, message: 'More than one person in video' },
];
