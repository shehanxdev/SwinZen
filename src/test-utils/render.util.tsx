import { NavigationContext } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import Config from 'react-native-config';
import { Provider as PaperProvider } from 'react-native-paper';
import { Provider } from 'react-redux';

import { paperTheme } from '@sz/config';
import { ConfigService } from '@sz/services';
import { store } from '@sz/stores';

type RenderParams = Parameters<typeof render>;

export function renderWithProviders(ui: RenderParams[0], options?: RenderParams[1]) {
  ConfigService.setCallback(key => Config[key]);

  const BaseProviders = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <NavigationContext.Provider
        //@ts-ignore
        value={{
          isFocused: () => true,
          addListener: jest.fn(() => jest.fn()),
        }}>
        <PaperProvider theme={paperTheme}>{children}</PaperProvider>
      </NavigationContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: BaseProviders, ...options });
}
