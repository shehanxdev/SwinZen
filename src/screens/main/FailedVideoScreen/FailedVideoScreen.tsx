import React from 'react';
import { View } from 'react-native';

import { Text } from '@sz/components';
import { tw } from '@sz/config';
import { TextAlignment, TextVariant } from '@sz/constants';

import { BaseScreen } from '../../components/BaseScreen';
import { VideoUploadErrorCard } from '../components';
import { dummyvideoUploadFailedReasons as videoAnalysisFailedReasons } from './dummyvideoUploadFailedReasons';

export function FailedVideoScreen() {
  return (
    <BaseScreen>
      <View testID="FailedVideoScreenTestId" style={tw`mx-4 mt-5`}>
        <VideoUploadErrorCard showFooterAndTextLink={false} />
        <View style={tw`mt-8`}>
          <Text variant={TextVariant.Body1Regular} textAlign={TextAlignment.Left}>
            Here's why your video has failed
          </Text>
          <View style={tw`ml-9 mr-10.5 mt-2.5`}>
            {videoAnalysisFailedReasons.map((reason, key) => {
              return (
                <Text
                  key={key}
                  variant={TextVariant.Body2Regular}
                  textAlign={TextAlignment.Left}>{`\u2022 ${reason}`}</Text>
              );
            })}
          </View>
        </View>
      </View>
    </BaseScreen>
  );
}
