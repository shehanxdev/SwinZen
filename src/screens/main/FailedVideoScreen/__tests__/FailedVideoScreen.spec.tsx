import * as React from 'react';

import { renderWithProviders } from '@sz/test-utils';

import { FailedVideoScreen } from '../FailedVideoScreen';

describe('Failed Video Screen', () => {
  const getRenderedScreen = () => renderWithProviders(<FailedVideoScreen />);

  it('should render failed screen ui screen correctly', () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });
  it('should find failed video screen screen via TestId', () => {
    const { getByTestId } = getRenderedScreen();
    const results = getByTestId('FailedVideoScreenTestId');
    expect(results).toBeTruthy();
  });
});
