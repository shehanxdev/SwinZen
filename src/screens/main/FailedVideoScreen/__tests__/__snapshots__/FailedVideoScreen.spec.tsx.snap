// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Failed Video Screen should render failed screen ui screen correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      [
        4279917603,
        4278984207,
        4278650631,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
        "position": "relative",
      }
    }
  >
    <Image
      resizeMode="cover"
      source={
        {
          "testUri": "../../../src/assets/images/grassBackground.jpg",
        }
      }
      style={
        {
          "opacity": 0.1,
          "position": "absolute",
        }
      }
    />
    <RCTSafeAreaView
      style={
        {
          "height": "100%",
          "paddingTop": 50,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          [
            {
              "flexBasis": "0%",
              "flexGrow": 1,
              "flexShrink": 1,
            },
            {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            {
              "flexGrow": 1,
            }
          }
          keyboardShouldPersistTaps="handled"
        >
          <View>
            <View
              style={
                {
                  "marginLeft": 16,
                  "marginRight": 16,
                  "marginTop": 20,
                }
              }
              testID="FailedVideoScreenTestId"
            >
              <View
                style={
                  {
                    "borderRadius": 10,
                    "height": 192,
                    "overflow": "hidden",
                  }
                }
              >
                <BVLinearGradient
                  colors={
                    [
                      1475762197,
                      3220592661,
                    ]
                  }
                  endPoint={
                    {
                      "x": 0.5,
                      "y": 1,
                    }
                  }
                  locations={
                    [
                      0,
                      1,
                    ]
                  }
                  startPoint={
                    {
                      "x": 0.5,
                      "y": 0,
                    }
                  }
                  style={
                    {
                      "borderColor": "rgba(246, 88, 21, 0.34)",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "flexBasis": "0%",
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "overflow": "hidden",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexBasis": "0%",
                        "flexGrow": 1,
                        "flexShrink": 1,
                        "gap": 1,
                        "justifyContent": "center",
                      }
                    }
                  >
                    <SvgMock
                      fill="#F65815"
                      height={54}
                      width={54}
                    />
                    <Text
                      style={
                        [
                          {
                            "color": "#F8F9FA",
                            "lineHeight": 24,
                          },
                          {
                            "fontFamily": "SourceSansPro-Regular",
                            "fontSize": 16,
                            "fontWeight": "normal",
                            "textAlign": "center",
                          },
                        ]
                      }
                    >
                      Video analysis failed!
                    </Text>
                    <View
                      style={
                        {
                          "marginTop": 4,
                        }
                      }
                    >
                      <Text
                        onPress={[Function]}
                        style={
                          [
                            {
                              "color": "#FFFFFF",
                              "lineHeight": 20,
                            },
                            {
                              "fontFamily": "SourceSansPro-Regular",
                              "fontSize": 16,
                              "fontWeight": "bold",
                              "textAlign": "center",
                              "textDecorationLine": "underline",
                            },
                          ]
                        }
                      />
                    </View>
                  </View>
                </BVLinearGradient>
              </View>
              <View
                style={
                  {
                    "marginTop": 32,
                  }
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#F8F9FA",
                        "lineHeight": 28,
                      },
                      {
                        "fontFamily": "SourceSansPro-SemiBold",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "textAlign": "left",
                      },
                    ]
                  }
                >
                  Here's why your video has failed
                </Text>
                <View
                  style={
                    {
                      "marginLeft": 36,
                      "marginRight": 36,
                      "marginTop": 10,
                    }
                  }
                >
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "SourceSansPro-Regular",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    • Too far away from golfer
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "SourceSansPro-Regular",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    • Too close to golfer
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "SourceSansPro-Regular",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    • Poor lighting. (Usually when indoors or at dusk outside.)
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "SourceSansPro-Regular",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    • Too much clutter in background (Usually indoors)
                  </Text>
                  <Text
                    style={
                      [
                        {
                          "color": "#F8F9FA",
                          "lineHeight": 24,
                        },
                        {
                          "fontFamily": "SourceSansPro-Regular",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "textAlign": "left",
                        },
                      ]
                    }
                  >
                    • More than one person in video
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </RCTSafeAreaView>
  </BVLinearGradient>
</View>
`;
